example/scripts/start_emulators.sh

```
@@ -0,0 +1,222 @@
+
+cleanup() {
+  echo -e "\n${YELLOW}🛑 Stopping emulators...${NC}"
+  
+  # Kill tracked processes
+  for pid in "${RUNNING_PIDS[@]}"; do kill "$pid" 2>/dev/null; done
+  
+  # Stop all running services
+  for service in "${RUNNING_SERVICES[@]}"; do
+    local emulator_file="$EMULATORS_DIR/${service,,}.sh"
+    if [ -f "$emulator_file" ]; then
+      source "$emulator_file"
```

Shellcheck (0.10.0)

[warning] ShellCheck can't follow non-constant source. Use a directive to specify location.

(SC1090)